gtsam/linear/NoiseModel.h

```diff
@@ -1069,8 +1018,7 @@ namespace gtsam {
       inline virtual Vector unwhiten(const Vector& /*v*/) const
       { throw std::invalid_argument("unwhiten is not currently supported for robust noise models."); }
       inline virtual double distance(const Vector& v) const
-      { return this->whiten(v).squaredNorm(); }
-      // TODO(mike): fold the use of the m-estimator residual(...) function into distance(...)
+      { return robust_->residual(v.norm()); }
```

Comment:
This is suspect. Where did the whiten go. If unit tests did not catch this, it means a unit test is missing :-) We should probably add unit tests for regular and some robust model, e.g. Huber.

Reply:
2nd

Reply:
So there was a unit test which was commented out. It had the same TODO note as this issue, so I uncommented it and made the changes, and it passed without issues. :slightly_smiling_face: 

But yes, I was on the fence wrt whether this should be a new PR. Thanks for clarifying this.